src/vtcomposite.cpp

```diff
@@ -539,4 +575,255 @@ Napi::Value composite(Napi::CallbackInfo const& info)
+            {
+                std::string temp;
+                tbuilder.serialize(temp);
+
+                // If the serialized buffer is an empty string, do not
+                // gzip compress it. This will lead to a non-zero byte string
+                // which can be perceived as a valid vector tile.
+                //
+                // Instead do nothing and return an empty, non-gzip-compressed buffer.
+                // If the user wants to handle empty tiles separately from non-empty
+                // tiles, they must check "buffer.length > 0" in the resulting callback.
+                if (!temp.empty())
+                {
+                    tile_buffer = gzip::compress(temp.data(), temp.size());
+                }
+            }
+            else
+            {
+                tbuilder.serialize(tile_buffer);
+            }
+        }
+        // LCOV_EXCL_START
+        catch (std::exception const& e)
+        {
+            SetError(e.what());
+        }
+        // LCOV_EXCL_STOP
+    }
+    std::vector<napi_value> GetResult(Napi::Env env) override
+    {
+        if (output_buffer_)
+        {
+            std::string& tile_buffer = *output_buffer_;
+            auto buffer = Napi::Buffer<char>::New(
+                env,
+                tile_buffer.empty() ? nullptr : &tile_buffer[0],
+                tile_buffer.size(),
+                [](Napi::Env env_, char* /*unused*/, std::string* str_ptr) {
+                    if (str_ptr != nullptr)
+                    {
+                        Napi::MemoryManagement::AdjustExternalMemory(env_, -static_cast<std::int64_t>(str_ptr->size()));
+                    }
+                    delete str_ptr;
+                },
+                output_buffer_.release());
+            Napi::MemoryManagement::AdjustExternalMemory(env, static_cast<std::int64_t>(tile_buffer.size()));
+            return {env.Null(), buffer};
+        }
+        return Base::GetResult(env); // returns an empty vector (default)
+    }
+
+    std::unique_ptr<InternationalizeBatonType> const baton_data_;
+    std::unique_ptr<std::string> output_buffer_;
+};
+
+Napi::Value internationalize(Napi::CallbackInfo const& info)
+{
+    // validate callback function
+    std::size_t length = info.Length();
+    if (length < 3 || length > 4)
+    {
+        Napi::Error::New(info.Env(), "expected buffer, language, options and callback arguments").ThrowAsJavaScriptException();
+        return info.Env().Null();
+    }
+    Napi::Value callback_val = info[length - 1];
+    if (!callback_val.IsFunction())
+    {
+        Napi::Error::New(info.Env(), "last argument must be a callback function").ThrowAsJavaScriptException();
+        return info.Env().Null();
+    }
+
+    Napi::Function callback = callback_val.As<Napi::Function>();
+
+    // validate buffer object
+    Napi::Value buf_val = info[0];
+    if (!buf_val.IsObject())
+    {
+        return utils::CallbackError("first argument must be buffer object", info);
```

Comment:
nit: suggest this becomes "... be a Buffer object" b/c we're talking about the Node "Buffer" object as opposed to a general buffer.
```suggestion
        return utils::CallbackError("first argument must be Buffer object", info);
```